test/test_cli_integration.py

```diff
@@ -1,67 +1,145 @@
+# -*- coding: utf-8 -*-
+
+"""
+CLI Integration tests
+
+This test file has two separate classes to cover two different client flows.
+The first class is for testing configurations of the CLI. The second
+assumes the client has been initialized.
+"""
 import json
 import os
+import shutil
 import tempfile
 import uuid
 import unittest
+from unittest.mock import patch
 
 import requests
 
 from .util.cli_helpers import integration_test, invoke_cli
 
-from conjur.version import __version__
+class CliIntegrationTestConfigurations(unittest.TestCase):
 
+    # *************** HELPERS ***************
 
-# Not coverage tested since integration tests doesn't run in
-# the same build step
-class CliIntegrationTest(unittest.TestCase): # pragma: no cover
-    REQUIRED_ENV_VARS = {
-        'CONJUR_ACCOUNT': 'account',
-        'CONJUR_AUTHN_LOGIN': 'user',
-        'CONJUR_AUTHN_API_KEY': 'password',
-    }
+    def setup_cli_params(self, env_vars, *params):
+        self.cli_auth_params = ['--debug']
+        self.cli_auth_params += params
 
-    HTTP_ENV_VARS = {
-        'CONJUR_HTTP_APPLIANCE_URL': 'url',
-    }
+        return self.cli_auth_params
 
-    HTTPS_ENV_VARS = {
-        'CONJUR_HTTPS_APPLIANCE_URL': 'url',
-    }
+    def print_instead_of_raise_error(self, error_class, error_message_regex, exit_code):
+        output = invoke_cli(self, self.cli_auth_params,
+            ['list'], exit_code=exit_code)
+        self.assertRegex(output, error_message_regex)
 
-    HTTPS_CA_BUNDLE_ENV_VAR = { 'CONJUR_CA_BUNDLE': 'ca-bundle' }
+    def evoke_list(self, exit_code=0):
+        return invoke_cli(self, self.cli_auth_params,
+            ['list'], exit_code=exit_code)
 
-    DEFINED_VARIABLE_ID = 'one/password'
+    # *************** INITIAL CONFIGURATION TESTS ***************
 
+    '''
+    Validates that the conjurrc was created on the machine 
+    '''
+    @integration_test
```

Comment:
TODO check certificate passed in again
if passed certificate, then we will not fetch (not print fp)
Conjurrc should have new certificate passed in